tests/rsync/rsync_spec.lua

```diff
@@ -181,6 +182,60 @@ describe("rsync", function()
                 end)
             end)
         end)
+
+        describe("hooks", function()
+            it("success, on_exit called, on_error not called", function()
+                setup(function()
+                    Res_code = -1
+                    Res_data = ""
+                    require("rsync").setup({
+                        on_exit = function(code, command)
+                            Res_code = code
+                        end,
+                        on_error = function(data, command)
+                            Res_data = data
+                        end,
+                    })
+
+                    vim.cmd.w()
+
+                    helpers.wait_sync()
+                    assert.equals(Res_code, 0)
+                    assert.equals(Res_data[1], "")
+
+                    -- restore default
+                    require("rsync").setup({ on_exit = function(_, _) end, on_error = function(_, _) end })
+                end)
+            end)
+
+            it("error, on_exit called, on_error called", function()
+                setup(function()
```

Comment:
Don't use setup here. It's not needed and you are overwriting the original file.

Reply:
done